bayes_kit/rhat.py

```diff
@@ -1,11 +1,88 @@
 import numpy as np
 from numpy.typing import NDArray, ArrayLike
+import scipy as sp
 
 FloatType = np.float64
 VectorType = NDArray[FloatType]
 SeqType = ArrayLike
 
 
+def split_chains(chains: list[SeqType]) -> list[SeqType]:
+    """
+    Return a list of the input chains split in half.  The result will be
+    a list twice as long as the input.  For example,
+    ```
+    >>> split_chains([[1, 2, 3, 4], [5, 6, 7]])
+    [[1, 2], [3, 4], [5, 6], [7]]
+    ```
+
+    Parameters:
+    chains: List of univariate Markov chains.
+
+    Returns:
+    List of input chains split in half.
+    """
+    return [list(arr) for chain in chains for arr in np.array_split(chain, 2)]
+
+
+def rank_chains(chains: list[SeqType]) -> list[SeqType]:
```

Comment:
I think I am going to need more help on the types here.  I can't piece together yours and @WardBrian's comments into actions.

Reply:
I suspect @WardBrian knows the numpy type system better than I do, so I'll defer to him on the typing question.

Upon further reflection, since the chains may not be of the same length, this probably has to stay a `list` of something, so my earlier comment was irrelevant.